packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -517,60 +517,71 @@ function TableColumnHeader(props) {
         {columnProps.allowsSorting &&
           <ArrowDownSmall UNSAFE_className={classNames(styles, 'spectrum-Table-sortedIcon')} />
         }
+        {props.children}
       </div>
     </FocusRing>
   );
 }
 
-function ResizableTableColumnHeader({column}) {
+function ResizableTableColumnHeader(props) {
+  let {column} = props;
   let ref = useRef();
-  let {state} = useTableContext();
+  let {state, columnState} = useTableContext();
   let formatMessage = useMessageFormatter(intlMessages);
+  let [isResizingAvailable, setIsResizingAvailable] = useState(false);
+  let [isHovered, setIsHovered] = useState(false);
+  let isResizing = columnState.getResizingColumn()?.key === column.key;
 
   const onMenuSelect = (key) => {
     switch (key) {
       case 'sort-asc':
         state.sort(column.key, 'ascending');
+        setIsResizingAvailable(false);
         break;
       case 'sort-desc':
         state.sort(column.key, 'descending');
+        setIsResizingAvailable(false);
         break;
       case 'resize':
-        // focusResizer, needs timeout so that it happens after the animation timeout for menu close
-        setTimeout(() => {
-          focusSafely(ref.current);
-        }, 360);
+        focusSafely(ref.current);
         break;
+      default:
+        setIsResizingAvailable(false);
     }
   };
   let allowsSorting = column.props?.allowsSorting;
   let items = useMemo(() => {
-    let options = {
-      sortAscending: allowsSorting && {
+    let options = [
+      allowsSorting ? {
         label: formatMessage('sortAscending'),
         id: 'sort-asc'
-      },
-      sortDescending: allowsSorting && {
+      } : undefined,
+      allowsSorting ? {
         label: formatMessage('sortDescending'),
         id: 'sort-desc'
-      },
-      resize: {
+      } : undefined,
+      {
         label: formatMessage('resizeColumn'),
         id: 'resize'
       }
-    };
-    return Object.keys(options).reduce((acc, key) => {
-      if (options[key]) {
-        acc.push(options[key]);
-      }
-      return acc;
-    }, []);
+    ];
+    return options;
   }, [allowsSorting]);
+  // if we're resizing another column, then hover shouldn't show the resizer of a different column
+  let showResizer = isResizingAvailable || (isHovered && !columnState.isResizingColumn) || isResizing;
 
   return (
     <>
-      <MenuTrigger>
-        <TableColumnHeader column={column} />
+      <MenuTrigger onOpenChange={(e) => setIsResizingAvailable(e)}>
```

Comment:
Whats the reason behind making the resizer visible on column menu open?

Reply:
if it's display none it can't receive focus, that said I've updated because our new code waits until menu unmount, but this may need to come back